extensions/positron-notebook-controllers/src/extension.ts

```diff
@@ -53,15 +52,15 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 	}
 
 	// Set the hasRunningNotebookSession context when the active notebook editor changes.
-	context.subscriptions.push(vscode.window.onDidChangeActiveNotebookEditor((editor) => {
-		const value = notebookSessionService.hasRunningNotebookSession(editor?.notebook.uri);
-		setHasRunningNotebookSessionContext(value);
+	context.subscriptions.push(vscode.window.onDidChangeActiveNotebookEditor(async (editor) => {
+		const session = editor && await getNotebookSession(editor.notebook.uri);
+		setHasRunningNotebookSessionContext(Boolean(session));
 	}));
 
 	// Set the hasRunningNotebookSession context when a session is started/shutdown for the active notebook.
 	context.subscriptions.push(notebookSessionService.onDidChangeNotebookSession((e) => {
-		if (e.notebookUri === vscode.window.activeNotebookEditor?.notebook.uri) {
-			setHasRunningNotebookSessionContext(!!e.session);
+		if (e.notebookUri.toString() === vscode.window.activeNotebookEditor?.notebook.uri.toString()) {
```

Comment:
Do we need to be careful about case sensitivity in the comparison?

Reply:
It's pretty extensively used by other extensions for `vscode.Uri` comparison and as a map key and the docs mention "normalization" so I assume it's safe